src/dc_msg.rs

```diff
@@ -432,22 +413,18 @@ pub unsafe fn dc_msg_set_location(
     (*msg).param.set_float(Param::SetLongitude, longitude);
 }
 
-pub unsafe fn dc_msg_get_timestamp(msg: *const dc_msg_t) -> i64 {
-    if msg.is_null() || (*msg).magic != 0x11561156i32 as libc::c_uint {
+pub unsafe fn dc_msg_get_timestamp(msg: &dc_msg_t) -> i64 {
+    if msg.magic != 0x11561156i32 as libc::c_uint {
         return 0;
     }
-    return if 0 != (*msg).timestamp_sent {
-        (*msg).timestamp_sent
+    return if 0 != msg.timestamp_sent {
+        msg.timestamp_sent
     } else {
-        (*msg).timestamp_sort
+        msg.timestamp_sort
     };
 }
 
-pub fn dc_msg_load_from_db<'a>(msg: *mut dc_msg_t<'a>, context: &'a Context, id: u32) -> bool {
-    if msg.is_null() {
-        return false;
-    }
-
+pub fn dc_msg_load_from_db<'a>(msg: &mut dc_msg_t<'a>, context: &'a Context, id: u32) -> bool {
```

Comment:
May make sense to change the return type to `dc_msg_t` that way this wouldn't need `dc_msg_emtpy`

Reply:
Reasonable.